### PR TITLE
This is my capstone project

### PR DESCRIPTION
**Pull request by Htetaungkyaw71**
# This is my final pull request for capstone project

- Installed some packages and setup react application
- Removed unnecessary files 
- Created two actions creators and reducer for this project.
- Used React-Router in this project.
- Added Homepage and Detail page
- Added some styles for Homepage and Detail page
- Added some icons from font awesome
- Installed react-paginate from external library
- Added pagination to the homepage 
- Added top three games in the homepage
- Added some detail information of the game in the detail page 
- Added mock testing for reducers.
- Added testing for components
- Added user-event testing for navbar components 
- Fix all linters errors
- Updated Readme file

- Home page
  - When the page loads, the web app shows the list of games that could be filtered out by some parameter, for example by game id.
  - Along with game id, this web app will display some additional information of the game that come from the API.
  - When the user clicks (or taps) on a game item, the application navigates to the details page.

- Details page
  - In the details page, the web app retrieves data from the API to show detailed data in the category.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- The project is a single-page application (SPA) built with React and Redux.
- The data retrieved from the API should be stored in the Redux store.
- I filter the data that I retrieve from the API using a Filter stateless component.
- Every page (the main page and the pages for each item) have a unique route within the SPA.
- The project have been deployed and accessible online.
- This is my presentation link - https://www.loom.com/share/ec5e412b7669413e9c6a72e4a2dc59a4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -259,6 +227,8 @@ If you like this project...
 
 I would like to thank...
 
+Inspiration - https://freetogame.com/
+
```
Comment (emmiiorji) [inline README.md]: You have rightly included an acknowledgement of the source of the API. Kindly give also appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).

File: src/components/Home.jsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
+/* eslint-disable react/prop-types */
```
Comment (emmiiorji) [inline src/components/Home.jsx]: Kindly avoid disabling linter checks inline. They represent coding standards that you share with your team and most of the time enforce best coding practices.

File: src/components/Card.jsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/* eslint-disable react/prop-types */
```
Comment (emmiiorji) [inline src/components/Card.jsx]: As mentioned above, kindly avoid disabling linter checks inline. They represent coding standards that you share with your team and most of the time enforce best coding practices.

File: src/components/Home.jsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                breakLabel="..."
+                nextLabel="&raquo;"
+                onPageChange={handlePageClick}
+                pageRangeDisplayed={2}
+                pageCount={pageCount}
+                previousLabel="&laquo;"
+                renderOnZeroPageCount={null}
+                className="pagination"
+              />
+            </div>
```
Comment (emmiiorji) [inline src/components/Home.jsx]: Your project design looks great. It is part of the [requirements](https://github.com/microverseinc/curriculum-react-redux/blob/5801583dd2ad132a9326d4bd429291d954e4e824/capstone/react_capstone.md?plain=1#L24) to include a feature that allows searching for items on the home page. Kindly include this feature.

File: src/__test__/Card.test.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    const tree = render(
+      <Provider store={store}>
+        <BrowserRouter>
+          <Card game={game} />
+        </BrowserRouter>
+      </Provider>,
+    );
+    expect(tree).toMatchSnapshot();
+  });
```
Comment (emmiiorji) [inline src/__test__/Card.test.js]: You have done great to include tests. Kindly ensure that all tests pass.

image

**Review by emmiiorji — CHANGES_REQUESTED** [review]
Hi @Htetaungkyaw71, 
# Status: Changes Required ♻️

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

A little bit more

## Good Highlights✔️

- Linter checks are passing
- Adhered to Gitflow
- Good design
- Good commit messages


## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mckent05 — APPROVED** [review]
Hi @Htetaungkyaw71, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 
- Your project looks good 👌 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**